cmd/destroy/cmddestroy.go

```diff
@@ -41,7 +42,8 @@ func GetDestroyRunner(provider provider.Provider, ioStreams genericclioptions.IO
 // DestroyCommand creates the DestroyRunner, returning the cobra command associated with it.
 func DestroyCommand(f cmdutil.Factory, ioStreams genericclioptions.IOStreams) *cobra.Command {
 	provider := provider.NewProvider(f)
-	return GetDestroyRunner(provider, ioStreams).Command
+	loader := manifestreader.NewManifestLoader(f)
```

Comment:
Why are we creating a new object here, when the `ManifestReader` can be retrieved from the existing `Provider`. This change will necessitate other changes in layers above this that have created different `Providers` and `ManifestReaders`.

Reply:
This separation will break the `DualDelegatingProvider` in kpt.

Reply:
The main reason of the separation is to make the boundary between the parsing the applying more clear. The apply library doesn't need to know the `Reader` interface. The caller of the library need to make sure the `InventoryInfo` parsed from the `Reader` work with the `provider`. More details [here](https://github.com/kubernetes-sigs/cli-utils/pull/270#discussion_r517788024).

Yes, this will break the current `DualDelegatingProvider`. That's expected since we are changing the interface. I'll update the kpt side to work with the new cli-utils library.

Reply:
Yeah, this will require some adjustments for users of cli-utils, but it makes a much clearer distinction between what is needed for the apply library (the Provider interface) and what is needed in the cobra commands (ManifestLoader).